website/tools/build_docs.js

```diff
@@ -132,32 +121,55 @@ const generateFinalMarkdown = (title, text, entityMap) => {
     return prettier.format(header + text, prettierConfig);
 };
 
+async function getExamplesFromRepo() {
+    await fs.emptyDir(EXAMPLES_DIR_NAME);
+    process.chdir(EXAMPLES_DIR_NAME);
+    const { body } = await httpRequest({
+        url: EXAMPLES_REPO,
+        json: true,
+    });
+    const builtExamples = await buildExamples(body);
+    await addExamplesToSidebars(builtExamples);
+}
+
+async function buildExamples(exampleLinks) {
+    const examples = [];
+    for (const example of exampleLinks) {
+        const responseStream = await httpRequest({
+            url: example.download_url,
+            stream: true,
+        });
+        try {
+            const pipeline = promisify(stream.pipeline);
+            await pipeline(responseStream, fs.createWriteStream(example.name));
+            const exampleName = example.name.split('.')[0];
+            examples.push(`examples/${exampleName.replace(/_/g, '-')}`);
+            console.log(`Rendering example ${example.name}`);
+        } catch (err) {
+            throw err;
+        }
+    }
+    return examples;
+}
+
+async function addExamplesToSidebars(examples) {
+    sidebars.examples = { Examples: examples };
+    fs.writeFileSync(path.join(__dirname, '..', 'sidebars.json'), JSON.stringify(sidebars, null, 4));
+    console.log('Saving examples to sidebars.json');
+}
+
+
 const main = async () => {
     /* input and output paths */
     const sourceFiles = path.join(__dirname, '..', '..', 'src', '**', '*.js');
-    const exampleFiles = path.join(__dirname, '..', '..', 'examples', '**', '*.js');
     const sourceFilesOutputDir = path.join(__dirname, '..', '..', 'docs', 'api');
     const typeFilesOutputDir = path.join(__dirname, '..', '..', 'docs', 'typedefs');
-    const exampleFilesOutputDir = path.join(__dirname, '..', '..', 'docs', 'examples');
 
     /* get template data */
     const templateData = await jsdoc2md.getTemplateData({ files: sourceFiles });
-    const exampleData = await jsdoc2md.getTemplateData({ files: exampleFiles });
 
     // handle examples
-    const examplePromises = exampleData.map(async (example) => {
-        const { description, meta: { filename, path: filepath, lineno } } = example;
-        const code = await readFileFromLine(path.join(filepath, filename), lineno);
-        const sep = '```';
-        const codeblock = `${sep}javascript\n${code}\n${sep}`;
-
-        const title = filename.split('.')[0].split('_').map(word => `${word[0].toUpperCase()}${word.substr(1)}`).join(' ');
-        const header = getHeader(title);
-        const markdown = prettier.format(`${header}\n${description}\n${codeblock}`, prettierConfig);
```

Comment:
It would be nice to keep the formatting of markdown by prettier, to make sure all examples look the same.

Reply:
Fixed - using readStreamToString to save markdown to variable and add formatting.